Improve Frontend development infra (#34)
DashAISoftware/DashAI: README.rst

Development
===========

To download and run the development version of DashAI, first, download the repository
and switch to the developing branch: : 

.. code:: bash

    $ git clone https://github.com/DashAISoftware/DashAI.git
    $ git checkout staging

Frontend
--------

.. warning:: 

    All commands executed in this section must be run 
    from `DashAI/front`. To move there, run:

    .. code:: 

        $ cd DashAI/front

Prepare the environment
~~~~~~~~~~~~~~~~~~~~~~~

1. `Install the LTS node version <https://nodejs.org/en>`_. 

2. Install `Yarn` package manager following the instructions located on the
   `yarn getting started <https://yarnpkg.com/getting-started>`_ page.

> 3. Install the project packages using yarn: 

Reviewer: ```suggestion
3. Install the project packages using yarn on `DashAI/front`: 
```

.. code:: bash

>     $ yarn install

Reviewer: ```suggestion
    $ cd DashAI/front && yarn install
```

Running the frontend
~~~~~~~~~~~~~~~~~~~~~~

> Launch the front-end development server by running the following command:

Reviewer: ```suggestion
Launch the front-end development server by running the following command on `DashAI/front`:
```

.. code:: bash

    $ yarn start

Linting and formatting
~~~~~~~~~~~~~~~~~~~~~~

The project uses as default linter `eslint <https://eslint.org/>`_ with 
the `react/recommended`, `standard-with-typescript`` and `prettier`` styles. 

> To run manually the linter, execute:

Reviewer: ```suggestion
To manually run the linter, execute on `DashAI/front`:
```

.. code:: bash

    $ yarn eslint src 

The project uses `prettier <https://prettier.io/>`_ as default formatter. 

To format the code manually, execute:

.. code:: bash
    
    $ yarn prettier --write src

Build the frontend
~~~~~~~~~~~~~~~~~~

> Execute:

Reviewer: ```suggestion
Execute from `DashAI/front`:
```

.. code:: bash

    $ yarn build

Backend
-------

.. warning:: 

    All commands executed in this section must be run 
    from `DashAI/back`. To move there, run:

    .. code:: 

        $ cd DashAI/back

Prepare the environment
~~~~~~~~~~~~~~~~~~~~~~~

First, set the python enviroment using 
`conda <https://docs.conda.io/en/latest/miniconda.html>`_:

.. code: bash

    $ conda create -n dashai python=3.10
    $ conda activate dashai 

and install the requirements: 

.. code:: bash

    $ pip install -r requirements.txt
    $ pip install -r requirements-dev.txt

Running the Backend
~~~~~~~~~~~~~~~~~~~

There are two ways to run DashAI:

> 1. By executing DashAI as a module:

Reviewer: ```suggestion
1. By executing DashAI as a module from the root directory:
```

If you chose the second way, remember to install it each time you make changes.

Execute tests
~~~~~~~~~~~~~

DashAI uses `pytest <https://docs.pytest.org/>`_ to perform the backend
tests.

> To execute the backend tests run:

Reviewer: ```suggestion
To execute the backend tests run from the root directory:
```

.. code:: bash

    $ pytest tests/

.. note:: 
    
    The database session is parametrized in every endpoint as 
    ``db: Session = Depends(get_db)`` so we can test endpoints on a test database 
    without making changes to the main database.

Linting and formatting
~~~~~~~~~~~~~~~~~~~~~~

> The project uses as default backend linter 

Reviewer: Where do we run the following code? `DashAI/back` or on root

Author: I removed the warnings and added a step to each instruction to indicate whether or not to move to each path.
Maybe later we should think about making a makefile.

`ruff <https://github.com/charliermarsh/ruff>`_:

> To run manually the linter, execute

Reviewer: ```suggestion
To manually run the linter, execute
```

.. code:: bash

    $ ruff .

> The project uses `prettier <https://prettier.io/>`_ as default formatter. 

Reviewer: ```suggestion
The project uses `black <https://black.readthedocs.io/en/stable/>`_ as default formatter. 
```

> To format the code manually, execute:

Reviewer: ```suggestion
To manually format the code, execute:
```
